star-dash/star-dash/GameEngine/Entities/EntityFactory.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  EntityFactory.swift
+//  star-dash
+//
+//  Created by Ho Jun Hao on 27/3/24.
+//
+
+import Foundation
+
+struct EntityFactory {
```

Comment:
This probably needs some discussion, but I was thinking if we should use a [Builder Pattern](https://www.geeksforgeeks.org/builder-design-pattern/) here?

Reply:
Builder pattern sounds good. Incorporated it.

But i think we should still have the factory pattern. The whole idea is that we don't want users to forget to add components into the game. Builder pattern still requires player to manually add the components.


Now when you call the factory, it will instantiate an EntityBuilder to create the components of the entity.